android-o/AutofillFramework/AutofillService/AutofillHintProperties.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AutofillService.Model;
+
+namespace AutofillService
+{
+    /**
+     * Holds the properties associated with an autofill hint in this Autofill Service.
+     */
+    public class AutofillHintProperties
+    {
+        private string mAutofillHint;
+        private IFakeFieldGenerator mFakeFieldGenerator;
+        private HashSet<int> mValidTypes;
+        private int mSaveType;
+        private int mPartition;
+
+        public AutofillHintProperties(string autofillHint, int saveType, int partitionNumber,
+            IFakeFieldGenerator fakeFieldGenerator, params int[] validTypes)
```

Comment:
Let's change the signature here to accept `SaveDataType saveType` and `params AutofillType[] validTypes` instead of ints, to avoid casting elsewhere.